lib/helpers.js

```diff
@@ -123,6 +125,17 @@ export const parseResults = (editor, results, filePath, showIgnored) => {
   return toReturn;
 };
 
+export const saveCleanedStyles = async (editor, results) => {
+  const result = results._postcssResult;
+  const styles = result.root.toString(result.opts.syntax);
+
+  if (styles === editor.getText()) {
+    // break to prevent infinite loop
```

Comment:
So looking over this, probably the simplest way to set this up is to change [this line](https://github.com/AtomLinter/linter-stylelint/blob/35ff0269e85f13210e74cf6b8510773b2cd899bf/lib/index.js#L200) to `await` on the result and store it in a variable. After that line grab the current editor contents and save it to `this.lastFixState` or something similar. Then return the results you just saved. To check the state against this change it so right after [this line](https://github.com/AtomLinter/linter-stylelint/blob/35ff0269e85f13210e74cf6b8510773b2cd899bf/lib/index.js#L89) it checks whether `text` is the same as `this.lastFixState` and if so it immediately does a `return null;`, which tells Linter to just keep any messages it currently has without updating anything.